kubectl-cloudflow/docker/client.go

```diff
@@ -147,53 +151,71 @@ func imageMatchesNameAndTag(image types.ImageSummary, imageNameAndTag string) bo
 	return false
 }
 
-// GetStreamletDescriptorsForImage extracts the configuration of a Cloudflow label from a docker image
+// GetStreamletDescriptorsForImageWithRegistryInspection extracts the configuration of a Cloudflow label from a docker image
 // and the image with digest in a struct
-func GetStreamletDescriptorsForImage(cli *client.Client, imageName string) (cloudflowapplication.CloudflowStreamletDescriptorsDigestPair, string, *PulledImage, error) {
-	// need this till we have a way to use docker registry API to check labels
-	pulledImage, pullError := PullImage(cli, imageName)	
-	if pullError != nil {	
-		return cloudflowapplication.CloudflowStreamletDescriptorsDigestPair{}, "", nil, fmt.Errorf("Failed to pull image %s: %s", imageName, pullError.Error())
-	}
-
-	images, err := cli.ImageList(context.Background(), types.ImageListOptions{})
+func GetStreamletDescriptorsForImageWithRegistryInspection(imageName string)(cloudflowapplication.CloudflowStreamletDescriptorsDigestPair, string, *PulledImage, error) {
+	ins := InspectOptions{Image: &ImageOptions{}}
+	localInspectOuput, err := ins.InspectLocalDockerImage(imageName)
```

Comment:
There can be a difference between the local image and the remote image, we should at least check that the sha is the same (since the image that is remote will actually be deployed)

Reply:
That means we will have to assume it exists at the remote registry. What if it does not exist and it has not been pushed yet, then we should allow verification anyway.

Reply:
That is correct. I think we need to think a bit about the UX. 
- What does it mean when you do "kubectl cloudflow verify", is it always remote, or, 
- only remote if local is not the same, or 
- can you pass in a flag --local or something.. 
- or do we use specific image refs "schemas", local://, and docker:// or something
- What if I want to runlocal, can I use my local images.. 
Something to discuss I think. (it's a not really part of this PR, maybe)
@agemooij

Reply:
As I've said in the various JIRa discussions, I do not see much use for using local images for anything besides the (future, as yet unimplemented) multi-image local running feature. We should always use the version on the registry to verify and deploy the blueprint.
AFAIK there is not a significant performance benefit to look at local images, especially not if we first have to check SHAs against the server anyway to make sure the local image isn't stale. 

So I propose we remove any code to perform local inspections and iterate from there.

Reply:
@agemooij there is also another reason to avoid it for now. Talking to the local daemon takes consistently too much time, check this (copying from the github actions build): 
```
dockerhub inspect took 347.708373ms
local inspect took 24.204437677s
```
I noticed the same on my machine. I have no idea yet why.

Reply:
Wow that's counter-intuitive..